templates/tower-project.yaml

```
@@ -465,30 +479,60 @@ Resources:
               - "s3:GetBucketLocation"
               - "s3:ListBucket"
               - "s3:GetObject"
-              - "s3:PutObject"
-              - "s3:PutObjectTagging"
               - "s3:DeleteObject"
-              - "s3:*MultipartUpload*"
             Resource:
               - !Sub "arn:aws:s3:::${TowerScratch}"
               - !Sub "arn:aws:s3:::${TowerScratch}/*"
+          - Sid: TowerForgeRoleS3WriteAccess
+            Effect: Allow
+            Principal:
+              AWS:
+                - !GetAtt TowerForgeServiceRole.Arn
+                - !GetAtt TowerForgeBatchHeadJobRole.Arn
+                - !GetAtt TowerForgeBatchWorkJobRole.Arn
+            Action:
+              - "s3:PutObject"
+              - "s3:PutObjectTagging"
+              - "s3:*MultipartUpload*"
+            Resource:
+              - !Sub "arn:aws:s3:::${TowerScratch}/work/*"
```

I just checked [the docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpuoverview.html#mpuAndPermissions) and the `s3:ListBucketMultipartUploads` action works on buckets, not objects. So, we need to include the bucket in the list of resources for this policy statement. 

```suggestion
              - !Sub "arn:aws:s3:::${TowerScratch}"
              - !Sub "arn:aws:s3:::${TowerScratch}/work/*"
```